src/EFCore/DbContext.cs

```diff
@@ -726,27 +726,24 @@ private void SetEntityState(InternalEntityEntry entry, EntityState entityState)
             }
         }
 
-        private async Task SetEntityStateAsync(
+        private Task SetEntityStateAsync(
             InternalEntityEntry entry,
             EntityState entityState,
             CancellationToken cancellationToken)
         {
             if (entry.EntityState == EntityState.Detached)
             {
-                await DbContextDependencies.EntityGraphAttacher.AttachGraphAsync(
+                return DbContextDependencies.EntityGraphAttacher.AttachGraphAsync(
```

Comment:
Change to expression-bodied method while we're at it?

Reply:
I may be misreading the code, but isn't the following possible?

```suggestion
        private Task SetEntityStateAsync(
            InternalEntityEntry entry,
            EntityState entityState,
            CancellationToken cancellationToken)
        => entry.EntityState == EntityState.Detached
            ? DbContextDependencies.EntityGraphAttacher.AttachGraphAsync(
                entry,
                entityState,
                forceStateWhenUnknownKey: true,
                cancellationToken: cancellationToken)
            : entry.SetEntityStateAsync(
                entityState,
                acceptChanges: true,
                forceStateWhenUnknownKey: entityState,
                cancellationToken: cancellationToken);
                cancellationToken: cancellationToken);
```

Reply:
but doesnt that make it less readable?

Reply:
I personally always use expressions where possible, it's definitely more readable to me. But maybe someone else can with in.

Reply:
The expression is fine. It is the ternary that i think makes it more difficult to grok. 

But either way. Isnt that out of scope for this change?

Reply:
@SimonCropp @roji Did we come to any conclusion on this?

Reply:
i gave up and moved on. sorry.